src/libgzx/gzm.c

```diff
@@ -545,6 +545,133 @@ gzm_cat_r (struct gz_macro *gzm, const struct gz_macro *gzm1, const struct gz_ma
     return 0;
 }
 
+int
+gzm_slice(struct gz_macro *output_gzm, const struct gz_macro *input_gzm, uint32_t frame_start, uint32_t frame_end) 
+{ 
+    // Each macro must have recorded at least one rng seed for this process to work and be within the bounds of the macro frames
```

Comment:
I'm not totally sure I follow why this restriction is needed? I think a user should be able to create a new macro that has no rng seeds, provided that they also supply a suitable state for the start frame that syncs rng?

If you wanted rng to be guaranteed correct without new states the restriction would have to be only allow splits on frames that have an rng seed saved, but this seems unnecessarily restrictive so I think it's best to just let the user deal with it.

Reply:
If you didn't have any RNG seeds, wouldn't something like copying dampe race not work due to random flames?

Reply:
Suppose I wanted to split a macro such that the starting frame doesn't have a saved rng seed. This is currently allowed provided there is at least one frame with a saved rng seed between start and end. This already desyncs rng as gz will not apply the saved rng seed if the rng value is different to what it expected, and it will be as the initial rng for the macro might be different. To work around this, the user has to create an appropriate starting state on the correct frame to use with the new macro.
So there are two options
- Either be more restrictive and only allow splitting on frames that have saved rng. I don't think this should be done since it would greatly limit the usefulness of this tool.
- Or be even less restrictive and allow a user to split in any way they like, assuming that they will provide an appropriate starting state for their choice of splitting.